doc/development.md

```diff
@@ -89,4 +89,205 @@
+listed below.
+
+#### Args: (or Arguments:)
+
+List each parameter by name. The type of the parameter may optionally follow
+the name, enclosed in parentheses, and then the description of the parameter
+separated by a colon. If the type is optional or can take multiple types,
+that can be specified with comma-separated lists or use
+[type hinting](https://docs.python.org/3/library/typing.html) syntax.
+If the description of the parameter is too long to
+fit in 79 characters, following hanging lines are indented with 4 extra
+spaces.
+
+#### Returns: (or Yields: for generators)
+
+Start with the type of the return value, followed by a colon and its semantics.
+If the function only returns None, this section is not required. It may also
+be omitted if the docstring starts with Return or Yield
+(e.g., `Return a boolean indicating if the folder has the right permissions.`),
+and the opening sentence is sufficient to describe the return value.
+
+#### Raises:
+
+List all exceptions that are relevant to the interface, followed by a
+description. Use a similar exception name + (type) + colon + space and hanging
+indent style as described in `Args:`.
+
+```python
+def example_function(arg_1, arg_2=None):
+    """
+    This is an example docstring. Start with summary explanation of the
+    function main purpose. Can expand multiple lines.
+
+    Args:
+        arg_1 (_type_): Long description can be split into
+            multiple lines with extra indentation.
+        arg_2 (_type_, optional): _description_. Defaults to None.
+
+    Returns:
+        _type_: _description_
+
+    Raises:
+        OSError: _description_
+    """
+```
+
+### Comment blocks and Inline comments
+
+Comments are accepted in tricky parts of the code that need extra explanation.
+A comment can expand through multiple lines before the operation starts.
+Comments should have normal punctuation and grammar. Start sentences with
+uppercase letter, end with periods, and use commas where necessary.
+Comments should never describe the code.
+
+Inline comments are allowed, but discouraged. Inline comments are acceptable
+for very short comments, but shall never expand through multiple lines.
+To improve legibility, inline comments should start at least 2 spaces away from
+the code. Multiple inline comments in successive lines should be vertically
+aligned.
+
+```python
+    _MULTIPATHD = cmdutils.CommandPath("multipathd",
+                                       "/usr/sbin/multipathd",  # Fedora, EL7
+                                       "/sbin/multipathd")      # Ubuntu
+
+    # List of multipath devices that should never be handled by vdsm. The
+    # main use case is to filter out the multipath devices the host is
+    # booting from when configuring hypervisor to boot from SAN. This device
+    # must have a special rule to queue I/O when all paths have failed, and
+    # accessing it in vdsm commands may hang vdsm.
```

Comment:
Shouldn't "vdsm" start with a capital letter?

Reply:
Yeah... I copied the comment from the project because it looked like a nice example of a proper multi-line comment, but didn't notice that. I'll fix it here in the example or remove the sentence entirely.

Reply:
I can still see "vdsm" rather than "Vdsm" here.